pkg/volume/glusterfs/glusterfs.go

```diff
@@ -435,6 +389,24 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 			glog.Infof("glusterfs: successfully mounted %s", dir)
 			return nil
 		}
+
+		// Give a try without `auto_unmount mount option, because
+		// it could be that gluster fuse client is older version and
+		// mount.glusterfs is unaware of `auto_unmount`.
+		// Use a mount string without `auto_unmount``
+		autoMountOptions := mountOptions
+		for i := len(autoMountOptions) - 1; i >= 0; i-- {
```

Comment:
True. Its just about the 'efficiency'. I do agree more readable approach is as shown above, let me convert to that. Thanks !